snapcraft/plugins/v2/cmake.py

```diff
@@ -28,6 +28,17 @@
     - cmake-parameters:
       (list of strings)
       parameters to pass to the build using the common cmake semantics.
+
+This plugin also interprets these specific build-environment entries:
+
+    - SNAPCRAFT_CMAKE_INSTALL_PREFIX
```

Comment:
I feel like the lines are a bit blurred between schema and the variables...

Just wondering - have you considered to instead set the default list of `cmake-parameters: ["-DCMAKE_INSTALL_PREFIX=/", "-DCMAKE_BUILD_TYPE=Release"] if unset?  It provides a path for the user to override it, with the documentation reinforcing their consideration if overriding?  Seems to me to cover the 90% case without complexity.

Reply:
I have, the problem with that is exactly the overriding, most of the user errors and complaints we had from field is the fact that either things cannot be replaced or things cannot be added. This solution seems to cater for both those counter complaints.

Reply:
And yes, the lines are a bit blurred, this was my main concern with exposing both environment variables and "options" in the first place.